src/generator.cs

```diff
@@ -3908,6 +3910,12 @@ public void GenerateMethodBody (MemberInformation minfo, MethodInfo mi, string s
 
 		CurrentMethod = String.Format ("{0}.{1}", type.Name, mi.Name);
 
+		// Warn about [Static] used in a member of [Category]
+		var baseTypeAtt = AttributeManager.GetCustomAttribute<BaseTypeAttribute> (minfo.type);
+		var staticAtt = AttributeManager.GetCustomAttribute<StaticAttribute> (mi);
```

Comment:
Use `HasAttribute` instead of getting all the attributes when you only need to check for attribute presence (it's slightly faster and allocates less memory).